nextjs/src/app/(app)/data-sources/create/connect/[externalDataSourceType]/page.tsx

```diff
@@ -505,6 +543,17 @@ export default function Page({
                     )}
                   />
                   <FPreopulatedSelectField name="geographyColumn" label={`${form.watch("geographyColumnType")?.toLocaleLowerCase()} field`} required />
+                  {testSourceResult.data?.testDataSource.crmType === CrmType.Editablegooglesheets && (
+                  <FPreopulatedSelectField 
+                    name="editablegooglesheets.idField" 
+                    label="Unique field" 
+                    required
+                    helpText={`
+                      Choose a column in your data that should always contain a unique value
+                      (e.g. email address for members, creation timestamp for events, etc.)
+                    `}
```

Comment:
Or "row count"? Would require special logic but might make sense for research sheets.